### PR TITLE
Kmsfs

### PR DESCRIPTION
**Pull request by maraino**
### Description
This PR adds an io.FS interface to KMSs to read public keys and certificates, it also adds a method to extract the crypto key from an ssh key. This PR includes other minor fixes and more test coverage.

'step-kms-plugin` commands use these changes.

### CODE REVIEW

File: kms/kmfs.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
+package kms
```
Comment (dopey) [inline kms/kmfs.go:1]: Is this file supposed to be named kmsfs.go?

File: kms/kmfs.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+// KeyFS creates a new KeyFS with the given KMS URI.
+func KeyFS(ctx context.Context, kmsuri string) (fs.FS, error) {
+	km, err := newFS(ctx, kmsuri)
+	if err != nil {
+		return nil, err
+	}
+	return &keyFS{kmsfs: km}, nil
+}
+
+// Open opens a file from a regular filesystem or a public key or certificate
```
Comment (dopey) [inline kms/kmfs.go:98]: Is the documentation here correct. Is it actually trying to open a file from a regular filesystem? Looks like it's just trying to open a public key / cert from the KMS.

File: sshutil/sshutil.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+		return CryptoPublicKey(sshPub)
+	case ssh.PublicKey:
+		// sk keys do not implement ssh.CryptoPublicKey
+		return cryptoSKPublicKey(p)
+	default:
+		return nil, fmt.Errorf("unsupported public key type %T", pub)
+	}
+}
+
+// cryptoSKPublicKey returns the crypto.Public key of an SSH SK public keys.
```
Comment (dopey) [inline sshutil/sshutil.go:37]: public key.

Comment (dopey) [reply sshutil/sshutil.go:37]: returns the crypto.PublicKey of an SSH SK public key.

**Review by dopey — APPROVED** [review]
left some comments that are mostly about documentation.

File: kms/kmfs.go (hunk 2 above)
Comment (maraino) [reply kms/kmfs.go]: Fixed with a676670

**Pull request merged**